shivyc/tree/expr_nodes.py

```diff
@@ -798,6 +798,48 @@ class PostDecr(_IncrDecr):
     return_new = False
 
 
+class _ArithUnOp(_RExprNode):
+    """Base class for unary plus and minus."""
+
+    descrip = None
+    cmd = None
+
+    def __init__(self, expr):
+        """Initialize node."""
+        super().__init__()
+        self.expr = expr
+
+    def make_il(self, il_code, symbol_table, c):
+        """Make code for this node."""
+        expr = self.expr.make_il(il_code, symbol_table, c)
+        if expr.ctype.is_arith():
+            # perform integer promotion
+            if expr.ctype.size < 4:
+                expr = set_type(expr, ctypes.integer, il_code)
+            if self.cmd:
+                out = ILValue(expr.ctype)
+                il_code.add(self.cmd(out, expr))
+                return out
+            else:
+                return expr
+        else:
+            err = f"wrong type argument to unary {self.descrip}"
```

Comment:
Nit: in order to be more specific to the user, could we specify what type we actually expected? Something like "unary plus requires arithmetic type operand"

Reply:
Yeah, I thought that was a little vague, but I was copying the error I got from gcc. I'll make it more specific. Thanks!